### DISCUSSION TITLE
[Implemented] further CMI /homes gui control

### DISCUSSION THREAD

**Discussion by mrfloris** [Zrips/CMI#6253] [Enhancement]
'sup,

Me and maybe a couple of others would love to see some extended options for controlling how the homes gui looks like. 

An example would be to extend the config.yml homes options to ..

- separate the /home bed icon.
- separate the /home home icon.
- option to include a button for death-location (if you have the permission to /dback or to teleport to it)
- option to add select alignment where left/right/center (and left matches current default, it can be default)
- option to allow editing of the home name/location/icon from within the homes gui
- option to use a custom item as the surrounding menu border, and
- option to set a custom item for 'empty space'
- if not yet possible, the ability to set the pagination icons (it's wool for me, can i change this to arrows, etc?)
- add a way to link the gui button to some command, so i can include ctext or openbook or some ctellraw link to forum thread. Giving us a way to educate players what they can (and cannot) do.
- Permission to allow a group "no matter what, THIS is the player's respawn point"
- separate the gui into public/private homes, so it's not a guessing game. 
- separate the gui into global, worlds, (and private..) homes. 
- Permission and ability to allow a player to set something like /home visit, then we can make a /visit command and a permission for others to go to /home visit bobby ..  (so one to set, one to use) and our custom cmi aliaseditor for /visit will be able to do the rest.
- If there's a command cost for visiting your home, some way of displaying this

Currently the home system is a chest interface with all the rows except the bottom being homes. And it auto grows as there's more homes. 

However, if we have permission to set a border, some dedicated buttons for home home and home bed, home visit, etc. And the ability to manage our homes from this main gui, .. then with a "buffer" around the border, and custom items .. we can have a dynamic and custom experience per server on how homes look like and it will be easier for players to understand what their options are.

Here's a dirty mockup to demonstrate some icons, with the help icon on the bottom left, pagination being custom, the center bottom is the home home, the bed is the home bed, and the banner is the home visit. 

<img width="406" alt="Screenshot 2021-10-12 at 22 36 08" src="https://user-images.githubusercontent.com/28841349/137025441-ad765a2c-46c8-46da-840c-e296a8199a23.png">

**Comment by NoltoxGit** [comment]
Very noiiice idea!

**Comment by mrfloris** [comment]
Most of these are now implemented;
https://www.spigotmc.org/resources/cmi-298-commands-insane-kits-portals-essentials-economy-mysql-sqlite-much-more.3742/update?update=427350
Upgrade CMILIB and CMI to get all the features mentioned in the changelog.
